datumaro/components/media.py

```diff
@@ -178,3 +179,255 @@ def save(self, path):
+            end_frame: Optional[int] = None) -> None:
+        super().__init__(path)
+
+        self._iterator = None
+        self._frame_size: Optional[Tuple[int, int]] = None
+
+        self._container = None
+        self._reset()
+
+        self._step = int(step)
+        self._start_frame = int(start_frame)
+        self._end_frame = int(end_frame) if end_frame else None
+
+        self._frame_count = self._get_frame_count(self._container)
+        self._length = None
+        self._get_length()
+
+    def close(self):
+        self._iterator = None
+
+        if self._container is not None:
+            self._container.release()
+            self._container = None
+
+    def __getitem__(self, idx: int) -> VideoFrame:
+        if not self._includes_frame(idx):
+            raise IndexError(f"Video doesn't contain frame #{idx}.")
+
+        return self._get_iterator()[idx]
+
+    def get_frame_data(self, idx: int) -> VideoFrame:
+        if not self._includes_frame(idx):
+            raise IndexError(f"Video doesn't contain frame #{idx}.")
+
+        return self._get_iterator().get_frame_data(idx)
+
+    def __iter__(self) -> Iterator[VideoFrame]:
+        """
+        Iterates over frames lazily, if possible.
+        """
+
+        if self._frame_count is not None:
+            # Decoding is not necessary to get frame pointers
+            # However, it can be inacurrate
+            end_frame = self._get_end_frame()
+            for index in range(self._start_frame, end_frame, self._step):
+                yield VideoFrame(video=self, index=index)
+        else:
+            # Need to decode to iterate over frames
+            yield from self._get_iterator()
+
+    @property
+    def frame_count(self) -> Optional[int]:
+        """
+        Returns frame count, if video provides such information.
+        Note that not all videos provide length / duration metainfo.
+
+        The count is affected by the frame filtering options of the object.
+        """
+        return self._get_length()
+
+    @property
+    def frame_size(self) -> Tuple[int, int]:
+        if self._frame_size is None:
+            self._frame_size = self._get_frame_size()
+        return self._frame_size
+
+    def _get_frame_size(self) -> Tuple[int, int]:
+        w = self._container.get(cv2.CAP_PROP_FRAME_WIDTH)
+        h = self._container.get(cv2.CAP_PROP_FRAME_HEIGHT)
+
+        if h and w:
+            frame_size = (int(h), int(w))
+        else:
+            image = next(self._get_iterator()).data
+            frame_size = image.shape[0:2]
+
+        return frame_size
+
+    @staticmethod
+    def _get_frame_count(container) -> Optional[int]:
+        # Not all videos provide length / duration metainfo
+        # Note that this information can be invalid or inaccurate
+        # due to variable frame rate
+        # https://stackoverflow.com/a/47796468
+        video_length = container.get(cv2.CAP_PROP_FRAME_COUNT)
```

Comment:
```suggestion
        frame_count = container.get(cv2.CAP_PROP_FRAME_COUNT)
```

No reason to introduce a synonym.

Reply:
Done